components/match2/wikis/wildrift/match_summary.lua

```diff
@@ -0,0 +1,364 @@
+---
+-- @Liquipedia
+-- wiki=wildrift
+-- page=Module:MatchSummary
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local CustomMatchSummary = {}
+
+local Class = require('Module:Class')
+local DisplayHelper = require('Module:MatchGroup/Display/Helper')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local ChampionIcon = require('Module:ChampionIcon')
+local Table = require('Module:Table')
+local ExternalLinks = require('Module:ExternalLinks')
+local String = require('Module:StringUtils')
+
+local MatchGroupUtil = Lua.import('Module:MatchGroup/Util', {requireDevIfEnabled = true})
+local MatchSummary = Lua.import('Module:MatchSummary/Base', {requireDevIfEnabled = true})
+local OpponentDisplay = Lua.import('Module:OpponentDisplay', {requireDevIfEnabled = true})
+
+local _MAX_NUM_BANS = 5
+local _NUM_CHAMPIONS_PICK = 5
+
+local _GREEN_CHECK = '[[File:GreenCheck.png|14x14px|link=]]'
+local _NO_CHECK = '[[File:NoCheck.png|link=]]'
+
+local _EPOCH_TIME_EXTENDED = '1970-01-01T00:00:00+00:00'
+
+-- Champion Ban Class
+local ChampionBan = Class.new(
+	function(self)
+		self.root = mw.html.create('div'):addClass('brkts-popup-mapveto')
+		self.table = self.root:tag('table')
+			:addClass('wikitable-striped'):addClass('collapsible'):addClass('collapsed')
+		self:createHeader()
+	end
+)
+
+function ChampionBan:createHeader()
+	self.table:tag('tr')
+		:tag('th'):css('width','40%'):wikitext(''):done()
+		:tag('th'):css('width','20%'):wikitext('Bans'):done()
+		:tag('th'):css('width','40%'):wikitext(''):done()
+	return self
+end
+
+function ChampionBan:banRow(banData, gameNumber, numberOfBans)
+	self.table:tag('tr')
+		:tag('td')
+			:node(CustomMatchSummary._opponentChampionsDisplay(banData[1], numberOfBans, false, true))
+		:tag('td')
+			:node(mw.html.create('div')
+				:wikitext(CustomMatchSummary._createAbbreviation{
+					title = 'Bans in game ' .. gameNumber,
+					text = 'Game ' .. gameNumber,
+				})
+			)
+		:tag('td')
+			:node(CustomMatchSummary._opponentChampionsDisplay(banData[2], numberOfBans, true, true))
+	return self
+end
+
+function ChampionBan:create()
+	return self.root
+end
+
+-- MVP Class
+local MVP = Class.new(
+	function(self)
+		self.root = mw.html.create('div'):addClass('brkts-popup-footer'):addClass('brkts-popup-mvp')
+		self.players = {}
+	end
+)
+
+function MVP:addPlayer(player)
+	if not Logic.isEmpty(player) then
+		table.insert(self.players, player)
+	end
+	return self
+end
+
+function MVP:create()
+	local span = mw.html.create('span')
+	span:wikitext(#self.players > 1 and 'MVPs: ' or 'MVP: ')
+	for index, player in ipairs(self.players) do
+		if index > 1 then
+			span:wikitext(', ')
+		end
+		span:wikitext('[['..player..']]')
+	end
+	self.root:node(span)
+	return self.root
+end
```

Comment:
Should we move this to /Base, or alternatively create a /Base/MVP, as we're repeating identical display code between wikis.
I have a feeling most team based games would like to have MVP support

Reply:
i think move it to /Base is good

Reply:
#741